packages/block-editor/src/components/colors-gradients/style.scss

```
@@ -77,18 +77,33 @@ $swatch-gap: 12px;
 	border-right: 1px solid $gray-300;
 	border-bottom: 1px solid $gray-300;
 
-	&.first {
+	// Identify the first instance as the item that is not a subsequent item to an
+	// item with the same class. The `~` selector is used instead of `+` to take
+	// into account any potential `ToolsPanelItem` placeholder item.
+	&:not(& ~ &) {
 		margin-top: $grid-unit-30;
 		border-top-left-radius: $radius-block-ui;
 		border-top-right-radius: $radius-block-ui;
 		border-top: 1px solid $gray-300;
 	}
 
-	&.last {
+	// Identify the last instance as an item without any subsequent siblings
+	// with the same class. The `~` selector is used instead of `+` to take
+	// into account any potential `ToolsPanelItem` placeholder item.
+	&:not(:has(~ &)) {
```

```suggestion
	// Identify the last visible instance as placeholder items will not have this class.
	&:nth-last-child(1 of &) {
```

Unfortunately, in Firefox (without flags enabled) neither the current selector or the proposed fallback match. 

Selector lists for the `nth-child` pseudo-classes appear to have better [coverage](https://caniuse.com/css-nth-child-of) and work well for me testing across browsers.

It also has the added benefit of avoiding `:has` entirely for now and being much easier to grok.